tests/unit/test_s3.py

```diff
@@ -482,38 +483,43 @@ class TestS3UtilsAsf:
     # path style format example: https://s3.{region}.localhost.localstack.cloud:4566/{bucket-name}/{key-name}
     # hosted style format example: http://{bucket-name}.s3.{region}localhost.localstack.cloud:4566/
     # region is optional in localstack
-    # the requested has been forwarded by the router, and S3_VIRTUAL_HOST_FORWARDED_HEADER has been added with the
-    # original host header
     def test_uses_virtual_host_addressing(self):
         addresses = [
-            ({"host": f"https://aws.{LOCALHOST}:4566"}, False),
-            ({"host": f"https://{LOCALHOST}.aws:4566"}, False),
-            ({"host": f"https://{LOCALHOST}.swa:4566"}, False),
-            ({"host": f"https://swa.{LOCALHOST}:4566"}, False),
-            ({"host": "https://bucket.s3.localhost.localstack.cloud"}, True),
+            ({"host": f"aws.{LOCALHOST}:4566"}, False),
+            ({"host": f"{LOCALHOST}.aws:4566"}, False),
+            ({"host": f"{LOCALHOST}.swa:4566"}, False),
+            ({"host": f"swa.{LOCALHOST}:4566"}, False),
+            ({"host": "bucket.s3.localhost.localstack.cloud"}, True),
             ({"host": "bucket.s3.eu-west-1.amazonaws.com"}, True),
-            ({"host": "https://s3.eu-west-1.localhost.localstack.cloud/bucket"}, False),
-            ({"host": "https://s3.eu-west-1.localhost.localstack.cloud/bucket/key"}, False),
-            ({"host": "https://s3.localhost.localstack.cloud/bucket"}, False),
-            ({"host": "https://bucket.s3.eu-west-1.localhost.localstack.cloud/key"}, True),
-            (
-                {
-                    "host": "https://bucket.s3.eu-west-1.localhost.localstack.cloud/key/key/content.png"
-                },
-                True,
-            ),
-            ({"host": "https://s3.localhost.localstack.cloud/bucket/key"}, False),
-            ({"host": "https://bucket.s3.eu-west-1.localhost.localstack.cloud"}, True),
-            ({"host": "https://bucket.s3.localhost.localstack.cloud/key"}, True),
+            ({"host": "s3.eu-west-1.localhost.localstack.cloud/bucket"}, False),
+            ({"host": "s3.localhost.localstack.cloud"}, False),
+            ({"host": "s3.localhost.localstack.cloud:4566"}, False),
+            ({"host": "bucket.s3.eu-west-1.localhost.localstack.cloud"}, True),
+            ({"host": "bucket.s3.localhost.localstack.cloud/key"}, True),
```

Comment:
Ah, right, that's true. According to [RFC 7230 / HTTP1.1](https://datatracker.ietf.org/doc/html/rfc7230#section-5.4) this header _must not_ contain a protocol.
This doesn't mean there aren't any weird S3 clients out there, but it seems reasonable to expect this.
You mentioned this regex might be used somewhere else. Is this still the case, or are you referring to some historical usages which have been refactored in the meantime?

Reply:
Historical usages, sorry, I've done a pass already so this shouldn't be a problem.

Normally, this shouldn't be an issue anyway because we already match on the `host` header with our router, and we didn't have an issue since v2. This new regex is basically the same as the router, so the functionality should be unchanged with v3.

Reply:
Yeah, I can see that. For me the question really just is if it breaks the legacy provider...